src/overlays/effects/ovl_Effect_Ss_Solder_Srch_Ball/z_eff_ss_solder_srch_ball.c

```diff
@@ -25,7 +25,7 @@ u32 EffectSsSolderSrchBall_Init(GlobalContext* globalCtx, u32 index, EffectSs* t
     this->update = EffectSsSolderSrchBall_Update;
     this->life = 100;
     this->rUnused = initParams->unused;
-    this->actor = initParams->linkDetected; // actor field was incorrectly used as a pointer to something else
+    this->actor = (Actor*)initParams->linkDetected; // actor field was incorrectly used as a pointer to something else
```

Comment:
This should stay imo: it's a genuine, horrible, programming mistake, and shouldn't be swept under the rug.